configs/cache_partitioning/cache_partitioning.py

```diff
@@ -0,0 +1,212 @@
+    """
+    Verify that Requested partitioning policy has been provided with a suitable
+    configuration
+    """
+    if args.policy == "way":
+        assert (
+            type(args.way_allocation) == int
+        ), "Need to supply Way Policy allocation"
+        assert (
+            args.way_allocation >= 0
+        ), "Way Policy needs positive number of ways"
+
+    if args.policy == "max_capacity":
+        assert (
+            type(args.capacity_allocation) == float
+        ), "Need to supply Max Capacity Policy allocation"
+        assert (
+            args.capacity_allocation >= 0
+        ), "Max Capacity Policy needs allocation in range [0, 1]"
+        assert (
+            args.capacity_allocation <= 1
+        ), "Max Capacity Policy needs allocation in range [0, 1]"
+
+
+def generatePartPolicy(args):
+    """
+    Generate Partitioning Policy object based on provided arguments
+    """
+    if args.policy == "way":
+        allocated_ways = []
+        for i in range(0, args.way_allocation):
+            allocated_ways.append(i)
+
+        allocation = WayPolicyAllocation(partition_id=0, ways=allocated_ways)
+        Policy = WayPartitioningPolicy(allocations=[allocation])
+
+    if args.policy == "max_capacity":
+        Policy = MaxCapacityPartitioningPolicy(
+            partition_ids=[0], capacities=[args.capacity_allocation]
+        )
+
+    return Policy
+
+
+def configSystem():
+    """
+    Configure base system and memory
+    """
+
+    system = System(membus=IOXBar(width=128))
+    system.clk_domain = SrcClockDomain(
+        clock=STATIC_OPTIONS.system_frequency,
+        voltage_domain=VoltageDomain(),
+    )
+
+    # add memory
+    mem_range = AddrRange("128MB")
+    system.mem_ranges = [mem_range]
+
+    # Memory configuration
+    config_mem(STATIC_OPTIONS, system)
+    system.mem_ctrls[0].latency = STATIC_OPTIONS.request_latency
+    system.mem_ctrls[0].bandwidth = STATIC_OPTIONS.mem_bandwidth
+
+    return system
+
+
+parser = argparse.ArgumentParser(
+    formatter_class=argparse.ArgumentDefaultsHelpFormatter
+)
+
+parser.add_argument(
+    "--policy",
+    default="way",
+    choices=["way", "max_capacity"],
+    help="This option defines which Cache Partitioning Policy to use for "
+    "the system cache",
+)
+
+parser.add_argument(
+    "--capacity-allocation",
+    type=float,
+    default=0.5,
+    help="The amount of the cache to partition to the default PartitionID "
+    "when using Max Capacity Cache Partitioning Policy in [0,1] range",
+)
+
+parser.add_argument(
+    "--way-allocation",
+    type=int,
+    default=4,
+    help="The number of ways in the cache to partition to the default "
+    "PartitionID when using Way Cache Partitioning Policy",
+)
+
+args = parser.parse_args()
+verifyPolicyParams(args)
+
+
+m5.ticks.setGlobalFrequency(STATIC_OPTIONS.system_frequency)
+system = configSystem()
+
+# create an L2 cache to sit between to enforce Partitioning policies
```

Comment:
L2?

Reply:
Thanks for pointing this out - the comment should just read 
```
# create a cache to sit between to enforce Partitioning policies
```